src/editors/textEditor.js

```diff
@@ -17,87 +17,6 @@ import EventManager from './../eventManager';
 import { KEY_CODES } from './../helpers/unicode';
 import { stopPropagation, stopImmediatePropagation, isImmediatePropagationStopped } from './../helpers/dom/event';
 
-const TextEditor = BaseEditor.prototype.extend();
-
-/**
- * @private
- * @editor TextEditor
- * @class TextEditor
- * @dependencies autoResize
- */
-TextEditor.prototype.init = function() {
-  const that = this;
-  this.createElements();
-  this.eventManager = new EventManager(this);
-  this.bindEvents();
-  this.autoResize = autoResize();
-  this.holderZIndex = -1;
-
-  this.instance.addHook('afterDestroy', () => {
-    that.destroy();
-  });
-};
-
-TextEditor.prototype.prepare = function(row, col, prop, td, originalValue, cellProperties, ...args) {
-  const previousState = this.state;
-
-  BaseEditor.prototype.prepare.apply(this, [row, col, prop, td, originalValue, cellProperties, ...args]);
-
-  if (!cellProperties.readOnly) {
-    this.refreshDimensions(true);
-
-    const {
-      allowInvalid,
-      fragmentSelection,
-    } = cellProperties;
-
-    if (allowInvalid) {
-      this.TEXTAREA.value = ''; // Remove an empty space from texarea (added by copyPaste plugin to make copy/paste functionality work with IME)
-    }
-
-    if (previousState !== EditorState.FINISHED) {
-      this.hideEditableElement();
-    }
-
-    // @TODO: The fragmentSelection functionality is conflicted with IME. For this feature refocus has to
-    // be disabled (to make IME working).
-    const restoreFocus = !fragmentSelection;
-
-    if (restoreFocus) {
-      this.instance._registerImmediate(() => this.focus());
-    }
-  }
-};
-
-TextEditor.prototype.hideEditableElement = function() {
-  this.textareaParentStyle.top = '-9999px';
-  this.textareaParentStyle.left = '-9999px';
-  this.textareaParentStyle.zIndex = '-1';
-  this.textareaParentStyle.position = 'fixed';
-};
-
-TextEditor.prototype.showEditableElement = function() {
-  this.textareaParentStyle.zIndex = this.holderZIndex >= 0 ? this.holderZIndex : '';
-  this.textareaParentStyle.position = '';
-};
-
-TextEditor.prototype.getValue = function() {
-  return this.TEXTAREA.value;
-};
-
-TextEditor.prototype.setValue = function(newValue) {
-  this.TEXTAREA.value = newValue;
-};
-
-TextEditor.prototype.beginEditing = function(...args) {
-  if (this.state !== EditorState.VIRGIN) {
-    return;
-  }
-
-  this.TEXTAREA.value = ''; // Remove an empty space from texarea (added by copyPaste plugin to make copy/paste functionality work with IME).
-  BaseEditor.prototype.beginEditing.apply(this, args);
-};
-
 const onBeforeKeyDown = function onBeforeKeyDown(event) {
```

Comment:
`onBeforeKeyDown` could be pulled into the editor class, similarly to [this example in the copyPaste plugin](https://github.com/handsontable/handsontable/blob/develop/src/plugins/copyPaste/copyPaste.js#L132).